cmd/bucket-handlers.go

```diff
@@ -1726,18 +1732,29 @@ func (api objectAPIHandlers) ResetBucketReplicationStateHandler(w http.ResponseW
 		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrReplicationNoMatchingRuleError), r.URL)
 		return
 	}
-	target := globalBucketTargetSys.GetRemoteBucketTargetByArn(ctx, bucket, config.RoleArn)
-	target.ResetBeforeDate = UTCNow().AddDate(0, 0, -1*int(days/24))
-	target.ResetID = mustGetUUID()
-	if err = globalBucketTargetSys.SetTarget(ctx, bucket, &target, true); err != nil {
-		switch err.(type) {
-		case BucketRemoteConnectionErr:
-			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErrWithErr(ErrReplicationRemoteConnectionError, err), r.URL)
-		default:
-			writeErrorResponseJSON(ctx, w, toAPIError(ctx, err), r.URL)
+	tgtArns := config.FilterTargetArns(
+		replication.ObjectOpts{
+			OpType:    replication.ResyncReplicationType,
+			TargetArn: arn})
+
+	var rinfo ResyncTargetsInfo
+	var target madmin.BucketTarget
+	for _, tgtArn := range tgtArns {
+		target = globalBucketTargetSys.GetRemoteBucketTargetByArn(ctx, bucket, tgtArn)
+		target.ResetBeforeDate = UTCNow().AddDate(0, 0, -1*int(days/24))
+		target.ResetID = resetID
+		rinfo.Targets = append(rinfo.Targets, ResyncTarget{Arn: tgtArn, ResetID: target.ResetID})
+		if err = globalBucketTargetSys.SetTarget(ctx, bucket, &target, true); err != nil {
+			switch err.(type) {
+			case BucketRemoteConnectionErr:
+				writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErrWithErr(ErrReplicationRemoteConnectionError, err), r.URL)
+			default:
+				writeErrorResponseJSON(ctx, w, toAPIError(ctx, err), r.URL)
+			}
+			return
```

Comment:
How will know which one succeeded and which one failed? @poornas this is similar in behavior to Multi-object delete - we cannot take multiple targets but reply back as if all target succeeded.

Reply:
It is more meaningful to allow reset by specific target rather than a blind reset of all targets - I've changed this handler to require target ARN for the reset. Made sure to leave this compatible with older mc client where arn was not required for the single target case.